File: API/Features/Reservations/ViewModels/ReservationListVM.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Classes;

namespace API.Features.Reservations {

    public class ReservationListVM {

        public Guid ReservationId { get; set; }
        public string FromDate { get; set; }
        public string ToDate { get; set; }
        public string BoatName { get; set; }
        public string BoatLoa { get; set; }
        public bool IsDocked { get; set; }
        public bool IsDryDock { get; set; }
        public bool IsAthenian { get; set; }
        public bool IsOverdue { get; set; }
        public bool IsCash { get; set; }
        public bool IsSurprise { get; set; }
        public List<ReservationBerthVM> Berths { get; set; }
        public string OwnerName { get; set; }
        public SimpleEntity PaymentStatus { get; set; }
        public SimpleEntity BoatType { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/classes/view-models/reservation-list-vm.ts
import { ReservationListBerthVM } from './reservation-list-berth-vm'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface ReservationListVM {

    reservationId: string
    fromDate: string
    toDate: string
    isDocked: boolean
    paymentStatus: SimpleEntity
    isDryDock: boolean
    isOverdue: boolean
    isGray: boolean
    boatName: string
    boatLoa: string
    ownerName: string
    berths: ReservationListBerthVM[]
    joinedBerths: string

}
